keras_retinanet/utils/anchors.py

```diff
@@ -80,17 +103,18 @@ def anchors_for_shape(
     if scales is None:
         scales = np.array([2 ** 0, 2 ** (1.0 / 3.0), 2 ** (2.0 / 3.0)])
 
-    # skip the first two levels
-    image_shape = np.array(image_shape[:2])
-    for i in range(pyramid_levels[0] - 1):
-        image_shape = (image_shape + 1) // 2
+    if model is None:  # guess shapes
+        image_shape = np.array(image_shape[:2])
+        image_shapes = [(image_shape + 2 ** x - 1) // (2 ** x) for x in pyramid_levels]
```

Comment:
Nice :)

Reply:
Perhaps a warning that it's guessing shapes to the user? As this would depend on model?

Reply:
Yes a warning would at least warn the user. But if the guess is not appropriate for the backbone, then the user would still need a method to fix it, preferably different than just changing this code until it works for them. I'm not sure there is any ideal way to do this, but if someone has it, I'm willing to make it happen :).